ruby/lib/google/protobuf.rb

```diff
@@ -50,6 +50,72 @@ class TypeError < ::TypeError; end
   rescue LoadError
     require 'google/protobuf_c'
   end
+
+  module Google
+    module Protobuf
+      module Internal
+        def self.infer_package(names)
+          # Package is longest common prefix ending in '.', if any.
```

Comment:
What does the package mean here? ruby package?
Does ruby package also include common parent messages?

Reply:
It is the protobuf package. This is explained more in the C code that calls this.